### PR TITLE
feature: obtain permissions, onboarding

### PR DESCRIPTION
**Pull request by ghost**
Awaiting public SDK release. The Podfile needs to reflect the released public SDK version.

### CODE REVIEW

File: SampleApp/OnboardingViewController.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    @IBAction func activityAuthTapped(_ sender: Any) {
+        activityButton.isEnabled = false
+        permissions.obtainMotionActivityPermission()
+    }
+
+    @IBAction func endOnboardingTapped(_ sender: Any) {
+        PersistenceLayer.isOnboardingOver = true
+        if let delegate = delegate {
+            delegate.onboardingDidEnd()
```
Comment (Asase12) [inline SampleApp/OnboardingViewController.swift]: just in one line: 
delegate?.onboardingDidEnd()

File: SampleApp/AppDelegate.swift [hunk 2]
```diff
@@ -6,42 +6,44 @@
...
     }
 
     private func initMainWindow() {
-        window = UIWindow(frame: UIScreen.main.bounds)
-        viewController = ViewController()
-        window?.rootViewController = viewController
+        if PersistenceLayer.isOnboardingOver {
+            let storyboard = UIStoryboard(name: "Main", bundle: nil)
+            let viewController = storyboard.instantiateViewController(withIdentifier: "MainViewController") as! MainViewController
+            window?.rootViewController = viewController
```
Comment (henriquenfaria) [inline SampleApp/AppDelegate.swift]: I'd extract this to a function

File: SampleApp/AppDelegate.swift [hunk 3]
```diff
@@ -6,42 +6,44 @@
...
+        } else {
+            let storyboard = UIStoryboard(name: "Onboarding", bundle: nil)
+            let viewController = storyboard.instantiateViewController(withIdentifier: "OnboardingViewController") as! OnboardingViewController
+            viewController.delegate = self
+            window?.rootViewController = viewController
```
Comment (henriquenfaria) [inline SampleApp/AppDelegate.swift]: I'd extract this to a function

File: SampleApp/MainViewController.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        guard let motionTag = appDelegate.motionTag else {
+            return
+        }
+        if trackingSwitch.isOn {
+            motionTag.start(withToken: PersistenceLayer.token)
+        } else {
+            motionTag.stop()
+        }
+    }
+
```
Comment (henriquenfaria) [inline SampleApp/MainViewController.swift]: Extra empty line

**Review by henriquenfaria — COMMENTED** [review]
@kian-motion-tag I guess we should disable the "End onboarding" button if the other buttons are not green. WDYT?
hmm, not sure, we can discuss this if it's something that helps the other devs.

File: SampleApp/OnboardingViewController.swift (hunk 1 above)
Comment (ghost) [reply SampleApp/OnboardingViewController.swift]: true 👍

File: SampleApp/MainViewController.swift (hunk 4 above)
Comment (ghost) [reply SampleApp/MainViewController.swift]: yes

**Review by Asase12 — APPROVED** [review]

**Review by henriquenfaria — APPROVED** [review]

**Pull request merged**